x/concentrated-liquidity/swaps_test.go

```diff
@@ -55,6 +56,15 @@ type SwapTest struct {
 }
 
 var (
+	// Allow 0.01% margin of error.
+	multiplicativeTolerance = osmomath.ErrTolerance{
+		MultiplicativeTolerance: sdk.MustNewDecFromStr("0.0001"),
+	}
+	// Allow additive margin equal of 1. It may stem from round up token in and rounding down
+	// tokenOut in favor of the pool.
+	oneAdditiveTolerance = osmomath.ErrTolerance{
```

Comment:
Does this tolerance not cause issues when we're dealing with large numbers? I believe our balancer and stableswap rounding guarantees are just purely `10^-8` multiplicative

Reply:
We actually first check multiplicative tolerance which is 10^-4.

If the multiplicative tolerance check fails because we operate on small values, then we fall back to an additive tolerance of 1 which we assume is fine due to rounding.

See this for details: 
https://github.com/osmosis-labs/osmosis/blob/a522eea21804350b620f91d9b6c282ac12c7d89e/x/concentrated-liquidity/swaps_test.go#L2731-L2744